lib/Transform/XTenMinimizeLiveTensors.cpp

```diff
@@ -209,6 +198,24 @@ class XTenMinimizeLiveTensorsPass
     }
   }
 
+  /// Checks for illegal dead code by traversing operations and
+  /// verifying if they are dead and any of their operands match any OpInfo.
+  /// Dead code can be legal if it is not connected to operation collection
+  /// through 'OpInfo' data structure.
+  bool hasIllegalDeadCode() {
+    return llvm::any_of(currFn.getBody().getOps(), [&](Operation &op) {
+      auto hasUses = !op.use_empty();
+      auto hasNoOperands = op.getNumOperands() == 0;
```

Comment:
Is this needed or would `return llvm::any_of(op.getOperands(), ...` anyway return `false` if there are no operands?

Reply:
That is a good question. I initially didn't have this, but some more robust tests were not passing. I will double check if it still works, because the issue may have been related to CMake not re-linking the library to the executable.

Reply:
An op can have a user that is unused, so indirectly dead. I think Amy op that uses the result of an op we are scheduling but is itself not an op we are scheduling would not be correctly handled by the code

Reply:
I just updated the pass with your suggestion. Now there is no `hasUse` and we only check if the operation if part of the chain of ops that are connected to the return operation.